src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DiagnosticReportStatus.java

```diff
@@ -0,0 +1,69 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import org.apache.commons.lang3.StringUtils;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * Example STS++UN'
+ */
+@EqualsAndHashCode(callSuper = false)
+@Builder
+@Getter
+public class DiagnosticReportStatus extends Segment {
+
+    private static final String KEY = "STS";
+
+    private final String detail;
+    private final @NonNull ReportStatusCode event;
```

Comment:
As before, if agreed:
```suggestion
    @NonNull
    private final ReportStatusCode event;
```